imap_processing/cli.py

```diff
@@ -826,10 +826,10 @@ def do_processing(
             science_files = dependencies.get_file_paths(source="idex")
             datasets = PacketParser(science_files[0]).data
         elif self.data_level == "l1b":
-            if len(dependency_list) > 3:
+            if len(dependency_list) > 2:
```

Comment:
If you are expecting an explicit number you should test against that number. Here and below.
```suggestion
            if len(dependency_list) != 2:
```